### ISSUE TITLE
Upload CSV modal isn't closable via the keyboard

### ISSUE THREAD

**Issue by noahmoss** [metabase/metabase#51920]
### Describe the bug

This modal isn't closable by pressing escape on the keyboard, which is an accessibility issue.

![Image](https://github.com/user-attachments/assets/bec42b38-6210-49fc-85f0-9b7598a16025)

### To Reproduce

1. Go to a collection on a fresh Metabase instance
2. Click the upload button to open the setup modal
3. Notice that it can only be closed by clicking the x in the corner

### Expected behavior

Should be closed when pressing escape

### Information about your Metabase installation

```JSON
Current master `fe092c1156` on firefox
```

### Severity

p3
